src/merkle/tiered_smt/mod.rs

```diff
@@ -176,6 +176,54 @@ impl TieredSmt {
         old_value
     }
 
+    // ITERATORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns an iterator over all inner nodes of this [TieredSmt] (i.e., nodes not at depths 16
+    /// 32, 48, or 64).
+    ///
+    /// The iterator order is unspecified.
+    pub fn inner_nodes(&self) -> impl Iterator<Item = InnerNodeInfo> + '_ {
```

Comment:
I think the advice provider will miss an API to get the current index of a given key/value pair. That would be required, for example, to insert new key/value pairs into the tree.

Reply:
We probably won't use this data structure in the advice provider directly. It will be used primarily to initialize `MerkleStore`. Having said that, it is very possible that this is currently missing some interface methods, and I'll update the interface as I work though the MASM implementation of TSMT.